yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -326,6 +326,14 @@ private[spark] class Client(
     distCacheMgr.setDistFilesEnv(env)
     distCacheMgr.setDistArchivesEnv(env)
 
+    // if spark.submit.pyArchives is in sparkConf, set PYTHONPATH to be passed
+    // on to the ApplicationMaster and the executors.
+    if (sparkConf.contains("spark.submit.pyArchives")) {
+      val archives = sparkConf.get("spark.submit.pyArchives")
+      env("PYTHONPATH") = archives
+      sparkConf.setExecutorEnv("PYTHONPATH", archives)
```

Comment:
is this going to overwrite other things possibly on the path or does it end up appending?